enhancements/installer/assisted-installer.md

```diff
@@ -0,0 +1,320 @@
+---
+title: assisted-installer
+authors:
+  - "@avishayt"
+  - "@hardys"
+  - "@dhellmann"
+reviewers:
+  - "@beekhof"
+  - "@deads2k"
+  - "@hexfusion"
+  - "@mhrivnak"
+approvers:
+  - "@crawford"
+  - "@abhinavdahiya"
+  - "@eparis"
+creation-date: 2020-06-09
+last-updated: 2020-06-10
+status: implementable
+see-also:
+  - "/enhancements/baremetal/minimise-baremetal-footprint.md"
+---
+
+# Assisted Installer
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+1. Where do the UI and assisted installer application run for
+   disconnected installations?
+
+   * It could run from a host booted with the live ISO. If that host
+     is the bootstrap host, then access to the UI and API would be
+     interrupted when the host reboots to join the cluster. If that
+     host is not part of the cluster initially then the minimum
+     hardware footprint for creating a cluster increases.
```

Comment:
> We would either need to redirect it to the OCP console that is on the 2 control plane nodes before the bootstrap reboots, or show the user instructions for connecting to that new location on their own. We've mentioned using a VIP, but that seems like another networking requirement for the user to come up with before building the cluster

Note that this is different from the current VM pivot, where we need a VIP because it's two different hosts (different MAC, thus a different DHCP lease and IP) - in this case we're talking about rebooting the *same* host, so it's highly likely that it's going to come back up with the same IP, either because there's a reservation or because the lease is still valid on the DHCP server?

Reply:
> Note that this is different from the current VM pivot, where we need a VIP because it's two different hosts (different MAC, thus a different DHCP lease and IP) - in this case we're talking about rebooting the _same_ host, so it's highly likely that it's going to come back up with the same IP, either because there's a reservation or because the lease is still valid on the DHCP server?

After the pivot there is no UI on that host.  We'll need to run it somewhere on the cluster.  Even if by chance that "somewhere" is the same host, the user should access it via some VIP and not go to the host IP directly.

Reply:
True, after the pivot the API will be available because it's a master, but if workers are specified then the OpenShift console won't be available until the ingress controller comes up on the workers.

I'm not sure depending on a VIP is unreasonable, since we've already got an implementation for that with the cluster-hosted keepalived enabled via the `baremetal` platform, but the ingress VIP never points at the bootstrap, so we'd still have to redirect the user after the bootstrapping is complete. Hmm.

Reply:
Didnt we agree already that the UI iso will run from the admin laptop? Also, not sure phase 3 should be in this scope.

Reply:
> Didnt we agree already that the UI iso will run from the admin laptop?

No, we didn't reach a conclusion yet

Reply:
Let's save the rest of this conversation for the public version of the pull request.